### PR TITLE
feat(comment-ticket): #CO-1802 rights for reopen ticket on a new comment

### PR DESCRIPTION
**Pull request by Mattox0**
## Describe your changes

When a ticket is "closed" and someone writes a comment, a specific right must be assigned for the ticket to re-open and get the status "open"

## Checklist tests

With an account that does not have the correct rights, write a comment in a "closed" ticket, it will not reopen.
With the right, do the same manipulation, the ticket opens

## Issue ticket number and link

[https://jira.support-ent.fr/browse/CO-1802](https://jira.support-ent.fr/browse/CO-1802)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [X] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by qpic-cgi — APPROVED** [review]

File: src/main/resources/public/ts/controllers/controller.ts [hunk 1]
```diff
@@ -557,16 +557,18 @@ export const SupportController: Controller = ng.controller('SupportController',
...
 			$scope.checkUpdateTicket($scope.editedTicket);
 			if(!$scope.editedTicket.processing){
 				return;
 			}
 
 			if ($scope.ticket.status == $scope.editedTicket.status
 				&& $scope.editedTicket.status == model.ticketStatusEnum.CLOSED
-				&& !!$scope.editedTicket.newComment)
+				&& !!$scope.editedTicket.newComment
+				&& await $scope.schoolHasSpecificWorkflow(WORKFLOW.REOPEN_TICKET_ON_COMMENT)) {
```
Comment (JinDouang) [inline src/main/resources/public/ts/controllers/controller.ts:569]: tu as une fonction asynchrone dans ta condition sans gestion d'erreur 
gère cette fonction asynchrone avant ton IF et prévois un cas d'erreur (`try catch`) si jamais ta fonction asynchrone ne fonctionne pas

**Review by JinDouang — CHANGES_REQUESTED** [review]

File: src/main/resources/public/ts/controllers/controller.ts (hunk 1 above)
Comment (Mattox0) [reply src/main/resources/public/ts/controllers/controller.ts:569]: @JinDouang Il y a un `try catch` à l'intérieur de la fonction `schoolHasSpecificWorkflow` qui gère déjà ça, je dois en remettre un quand même ?

Comment (JinDouang) [reply src/main/resources/public/ts/controllers/controller.ts:569]: oui car c'est cette méthode updateTicket qu'il faudra gérer si ça fail

Comment (qpic-cgi) [reply src/main/resources/public/ts/controllers/controller.ts:569]: aux vu de ça: https://github.com/OPEN-ENT-NG/support/blob/7d63b3250b564837701cd8e47cb9e1db536ac3f9/src/main/resources/public/ts/controllers/controller.ts#L267

aprioris ça ne peut pas fail.

ça a déjà été utilisé comme tel ici:
https://github.com/OPEN-ENT-NG/support/blob/7d63b3250b564837701cd8e47cb9e1db536ac3f9/src/main/resources/public/ts/controllers/controller.ts#L277

**Pull request merged**